src/DotNet.Status.Web.Tests/DependencyRegistrationTests.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.Internal.DependencyInjection.Testing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Moq;
+using Xunit;
+
+namespace DotNet.Status.Web.Tests
+{
+    public class DependencyRegistrationTests
+    {
+        [Fact]
+        public void AreDependenciesRegistered()
+        {
+            Type[] siteTypes = typeof(Startup).Assembly.GetTypes();
+            var controllers = siteTypes.Where(t => typeof(ControllerBase).IsAssignableFrom(t)).ToArray();
+            Assert.True(DependencyInjectionValidation.IsDependencyResolutionCoherent(s =>
+                    {
```

Comment:
Just curious,It Asserts True, if all the thing in the parenthesis gets executed without any exception?

Reply:
Assert.True just asserts that the value is true.  The value in this case, is the return value from `DependencyInjectionValidation.IsDependencyResolutionCoherent`.  So it's essentially asserting Assert.IsDependencyResolutionCoherent.

What IsDependencyResolutionCoherent does is run the action you give it to configure a ServiceCollection, and then makes sure that everything in that collection (as well as, now, any extra types passed as the third parameter) has all it's dependencies it needs resolvable in the collection.

This detected the changes necessary in the Startup, namely that "GwtTokenProvider" didn't have things it needed (but wasn't actually used, so I deleted it, rather than add it's dependencies), as well as that there was no IGitHubClientFactory provided (the concrete type was registered as the service instead of the interface)